typespec-extension/src/utils.ts

```diff
@@ -60,6 +74,86 @@ export function pushDistinct<T>(targetArray: Array<T>, ...items: Array<T>): Arra
   return targetArray;
 }
 
+export function logWarning(program: Program, msg: string) {
+  program.trace("typespec-java", msg);
+  program.reportDiagnostic({
+    code: "typespec-java",
+    severity: "warning",
+    message: msg,
+    target: NoTarget,
+  });
+}
+
+export async function loadExamples(program: Program, options: EmitterOptions): Promise<Map<Operation, any>> {
+  const operationExamplesMap = new Map<Operation, any>();
+  if (options["examples-directory"]) {
+    const operationIdExamplesMap = new Map<string, any>();
+
+    const service = ignoreDiagnostics(getAllHttpServices(program))[0];
+    let version = undefined;
+    const versioning = getVersion(program, service.namespace);
+    if (versioning && versioning.getVersions()) {
+      const versions = versioning.getVersions();
+      version = versions[versions.length - 1].value;
+    }
```

Comment:
Processing here is a bit simplistic. It uses the last version (likely fine) of the first service (usually fine).